### PR TITLE
all classes of model side

### PR DESCRIPTION
**Pull request by guyshakk**
Added all model classes

### CODE REVIEW

File: src/Rubric.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+public class Rubric implements IRubric, Finals {
```
Comment (OrenZak) [inline src/Rubric.java]: check if we need Finals

File: src/TicTacToeBoard.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import java.util.ArrayList;
+
+public class TicTacToeBoard implements IBoard, Finals, IRegistrable {
+
+	private IRubric[] rubrics = new Rubric[BOARD_SIZE];
+	private IPlayer currentPlayer;
+	private CompareRubrics<IRubric> rubricComparator = new CompareRubrics<>();
+	private ArrayList<TicTacToeListener> listeners = new ArrayList<TicTacToeListener>();
+	
+	public TicTacToeBoard(IPlayer currentPlaye) {
```
Comment (OrenZak) [inline src/TicTacToeBoard.java]: var name `currentPlaye` -> `currentPlayer`

Comment (OrenZak) [inline src/TicTacToeBoard.java]: add more player, `currentPlayer` and `otherPlayer`

File: src/TicTacToeEvent.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+public class TicTacToeEvent implements Finals {
+
+	private TicTacToeRequests request;
+	private String[] values;
+	
+	public TicTacToeEvent(TicTacToeRequests request, String[] values){
```
Comment (OrenZak) [inline src/TicTacToeEvent.java]: create more constructor with no string[]

File: src/TicTacToeBoard.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import java.util.ArrayList;
+
+public class TicTacToeBoard implements IBoard, Finals, IRegistrable {
+
+	private IRubric[] rubrics = new Rubric[BOARD_SIZE];
```
Comment (OrenZak) [inline src/TicTacToeBoard.java]: array[][]

File: bin/.gitignore [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/TicTacToeApp.class
```
[the remark this answers is not in the document]
Comment (guyshakk) [reply bin/.gitignore]: 👍

File: src/Rubric.java (hunk 1 above)
Comment (guyshakk) [reply src/Rubric.java]: 👍

File: src/TicTacToeBoard.java (hunk 2 above)
Comment (guyshakk) [reply src/TicTacToeBoard.java]: 👍

Comment (guyshakk) [reply src/TicTacToeBoard.java]: 👍

File: src/TicTacToeEvent.java (hunk 3 above)
Comment (guyshakk) [reply src/TicTacToeEvent.java]: 👍

File: src/TicTacToeBoard.java (hunk 4 above)
Comment (guyshakk) [reply src/TicTacToeBoard.java]: 👍

**Pull request merged**